snuba/utils/registered_class.py

```diff
@@ -0,0 +1,70 @@
+from abc import ABCMeta
+from typing import Any, Dict, Tuple, Type, cast
+
+
+class NoConfigKeyError(Exception):
+    pass
+
+
+class InvalidConfigKeyError(Exception):
+    pass
+
+
+class _ClassRegistry:
+    """Keep a mapping of classes to their names"""
+
+    def __init__(self) -> None:
+        self.__mapping: Dict[str, RegisteredClass] = {}
+
+    def register_class(self, cls: "RegisteredClass") -> None:
+        key = cls.config_key()
+        existing_class = self.__mapping.get(key)
+        if not existing_class:
+            self.__mapping[key] = cls
```

Comment:
I actually think we should fail hard if a mapping already exists, I could see this being very confusing